packages/foreigntypes/posixtypes.pony

```diff
@@ -0,0 +1,95 @@
+// This file defines the standard Posix type to make FFI calls easier
+// to manage and more consistent.
+
+type BlkCnt is I64
+
+// not consistent across OSX (always 32 bit int)/Linux (Long)
+type BlkSize is ILong
+type OSXBlkSize is I32
```

Comment:
@sylvanc - if we can eliminate dead `ifdef` branches eagerly (before the `names` pass), it doesn't seem like this would add any additional work for the type checker, and would be quite simple to handle, but maybe I'm missing something.

The main barrier to this that I'm seeing is that it may be difficult to integrate this strategy with the parser.  Because the Pony parser is LL1, and type entity declarations and method bodies are parsed without a ending token, Pony relies on seeing the next `fun`, `new`, `class`, `actor`, `primitive`, `type`, or `struct` token to know that current method body is finished (or type entity declaration) is finished.  If we wanted to allow `ifdef`s at the top level scope of a file governing the selective compilation of types (or for that matter, within a type entity declaration governing the selective compilation of methods), we'd have to figure out something clever with the parser and/or the `syntax` pass that follows the parser to make it work.

One slightly ugly, but quite easy option would be to introduce a new keyword, (for example, `typeifdef`) for `ifdef`s that need to be used at the top level scope.  I don't really like this option, but I wanted to mention it.

Another option would be to let an `if` follow on the same line as the type, just as we do for `use` calls.
```pony
type BlkSize is ILong if linux
type BlkSize is I32 if osx
```